src/AccountServices.js

```diff
@@ -0,0 +1,257 @@
+ * @param  {MatrixTwitter}   opts.twitter
+ * @param  {object} opts.app_auth OAuth authentication information
+ * @param  {string} app_auth.consumer_key Twitter consumer key
+ * @param  {string} app_auth.consumer_secret Twitter consumer secret
+ */
+var AccountServices = function (opts) {
+  this._bridge = opts.bridge;
+  this._app_auth = opts.app_auth;
+  this._storage = opts.storage;
+  this._twitter = opts.twitter;
+  this._oauth = new OAuth.OAuth(
+    'https://api.twitter.com/oauth/request_token',
+    'https://api.twitter.com/oauth/access_token',
+    this._app_auth.consumer_key,
+    this._app_auth.consumer_secret,
+    '1.0A',
+    "oob",
+    'HMAC-SHA1'
+  );
+};
+
+/**
+ * processInvite - Handler for invites from a matrix user to a (presumably)
+ * empty room. This will join the room and send some help text.
+ * @param  {MatrixEvent} event   The event data of the request.
+ * @param  {Request} request The request itself.
+ * @param  {Context} context Context given by the appservice.
+ */
+AccountServices.prototype.processInvite = function (event, request, context) {
+  log.info("Handler.AccountServices", "Got invite");
+  var intent = this._bridge.getIntent();
+  intent.join(event.room_id).then( () => {
+    var rroom = new RemoteRoom("service_"+event.sender);
+    rroom.set("twitter_type", "service");
+    this._bridge.getRoomStore().linkRooms(context.rooms.matrix, rroom);
+    intent.sendMessage(event.room_id,
+      {
+        "body": "This bot can help you link/unlink your twitter account with the " +
+        "Matrix Twitter Bridge. If this was not your intention, please kick the bot.",
+        "msgtype": "m.text"
+      }
+    );
+    //Add the room to the list.
+  }).catch(err => {
+    log.error("Handler.AccountServices", "Couldn't join service room. %s", err)
+  });
+};
+
+AccountServices.prototype.processLeave = function (event, request, context) {
+  log.info("Handler.AccountServices", "User %s left room. Leaving", event.sender);
+  var intent = this._bridge.getIntent();
+  intent.leave(event.room_id).then(() =>{
+    var roomstore = this._bridge.getRoomStore();
+    roomstore.removeEntriesByRemoteRoomData(context.rooms.remote.data);
+  });
+}
+
+/**
+ * processMessage - Processing incoming commands from the matrix user.
+ * @param  {MatrixEvent} event   The event data of the request.
+ */
+AccountServices.prototype.processMessage = function (event) {
+  if(event.content.body == "link account") {
+    this._beginLinkAccount(event);
+  }
+  else if(event.content.body == "unlink account") {
+    this._unlinkAccount(event);
+  }
+  else if(isStrInteger(event.content.body)) {
+    this._processPIN(event);
+  }
+};
+
+
+/**
+ * _beginLinkAccount - Processes a users request to link their twitter account
+ * with the bridge. This should return a authorisation link.
+ *
+ * @param  {object} event The Matrix Event from the requesting user.
+ */
+AccountServices.prototype._beginLinkAccount = function (event) {
```

Comment:
You can tidy up the function signature here. You only ever use `event.sender` and `event.room_id`, so why not just accept those as parameters?